adapter/internal/oasparser/model/open_api.go

```diff
@@ -29,6 +30,10 @@ import (
 	logger "github.com/wso2/product-microgateway/adapter/internal/loggers"
 )
 
+const (
+	hostNameValidator = "^[a-zA-Z0-9]+[a-zA-Z0-9-.]+[0-9a-zA-Z]$"
+)
```

Comment:
Can you add a comment explaining the regex and with few samples, and few wrong samples

Reply:
I think this is the specification 
https://datatracker.ietf.org/doc/html/rfc952

Reply:
Comment can find here : 018832e